buildSrc/src/main/java/GithubDetailsTask.kt

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+/**
+ * Helper to write to the "customCheckRunText.md" file for Taskcluster.
+ * Taskcluster uses this file to populate the "Details" section in the GitHub Checks panel UI.
+ */
+open class GithubDetailsTask : DefaultTask() {
+
+    /**
+     * Text to write to the file.
+     * Links are automatically rewritten to point to the correct Taskcluster URL.
+     */
+    @Input
+    var text: String = ""
+
+    private val detailsFile = File("/builds/worker/github/customCheckRunText.md")
+    private val suffix = "\n\n_(404 if compilation failed)_"
```

Comment:
or if paths of anything changed, right? i'd remove this.

Reply:
It's unlikely for paths to change without adjusting gradle, but compilation failure can happen on any PR. I'd prefer to keep this for now and [switch to redirecting to taskcluster](https://github.com/mozilla-mobile/fenix/issues/12785) in the future.